README.md

```diff
@@ -69,7 +69,7 @@ or your package manager on Linux.
 5. Launch Neolink from a shell, passing your configuration file:
 
     ```bash
-    neolink --config my_config.toml
```

Comment:
I was thinking of that. We need to make the subcommands optional and have an optional `--config` that should only be `Some` when `subcommand` is `None`

Or... we could have one top level config

```
neolink --config=config.toml subcommand
```

But this means that each sub command has to have its config merged with the top level config

Reply:
So I am trying to get this to work but it is being stubborn. I think perhaps it will require the beta clap to do.

I can do it if I handle the case manually but that means we can't get claps nice error messages when they forget the `--config` or the subcommand.

I was originally thinking this

```rust
#[structopt(short, long, parse(from_os_str),required_unless("cmd"),conflicts_with("cmd"))]
pub config: Option<PathBuf>,
#[structopt(subcommand)]
pub cmd: Option<Command>,
```
But the `required_unless` and `conflicts_with` don't seem to work with sub commands. I will keep trying

Reply:
I have this working, not perfectly as I'd like to do this from within clap. But it is does what we want